src/test/shell/bazel/skylark_git_repository_test.sh

```diff
@@ -158,6 +161,27 @@ EOF
   expect_log "Pluto is a dwarf planet"
 }
 
+# This test:
+# 1. Creates a git_repository rule with some commit hash
+# 2. Make sure worksapce can be built and repository cache is updated
+function test_git_repositry_cache_is_populated() {
+  local pluto_repo_dir=$TEST_TMPDIR/repos/pluto
+  local cache_dir=/tmp/bazel_cache
+  local commit_hash="52f9a3f87a2dd17ae0e5847bbae9734f09354afd"
+
+
+  export BAZEL_GIT_REPOSITORY_CACHE=$cache_dir
+
+  # Do not run tests if the git worktree command does not exist.
+  # Testing without if check - is windows timing out due to that?
+  # if git worktree --help; then
+  do_git_repository_test $commit_hash
+
+  cache_commit_hash="$(cat $cache_dir/worktrees/pluto/HEAD)"
+  assert_equals $commit_hash $cache_commit_hash
```

Comment:
A verification of caching usually verifies that we can do a clone from cache only in an "offline" situation, if we successfully filled the cache first, and attempt to check out a fixed hash (not a branch). I don't see your test verifying that main property of a cache.